### PR TITLE
Don't replace the MIME type of the requested files.

### PR DESCRIPTION
**Pull request by mandrigin**
TL;DR
We tried to inject our web3.js even to binary files, also we replaced their MIME to `text/html`.

Longer:
We had two copy-paste code path (one for ServiceWorkers and one for WebView) that did the same.
For some reason, in the Service Workers' copy of that, we had a filter that we only inject JS to HTML files.
In the WebView's copy we just went ahead and injected it everywhere, into images, etc. And with doing that, we also replaced it's MIME type to `text/html` regardless of what it was before.

What I did:
1) Merged these code paths;
2) Made JS injection working only for the files that had `text/html` MIME anyways;
3) Only set MIME to `text/html` for the files that already were `text/html`.

### CODE REVIEW

**Review by janherich — APPROVED** [review]

**Review by yenda — APPROVED** [review]
well described summary!

**Comment by mandrigin** [comment]
Let's wait with merging it until 
https://github.com/status-im/status-react/pull/5436 is tested.

**Comment by mandrigin** [comment]
@Serhy tested it, looks fine, so I'll merge.
